core/src/services/azdls/crc64.rs

```diff
@@ -0,0 +1,287 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/// Computes CRC64 based on ADLS continuation token requirements
+/// Uses the polynomial 0x9a6c9329ac4bc9b5
+/// See [here](https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/utils/CRC64.java).
+pub(super) fn compute(input: &[u8]) -> u64 {
+    let mut value = !0u64;
+    for &byte in input {
+        value = TABLE[((byte as u64) ^ value) as usize & 0xFF] ^ (value >> 8);
+    }
+    !value
+}
+
+// Generated with poly 0x9a6c9329ac4bc9b5
```

Comment:
It's unlikely for us to touch code here, but it's must be better if we have a unit for them like test what's will be generated with input `data/hello_world.pdf`.